src/workerd/io/worker-interface.c++

```diff
@@ -205,6 +207,27 @@ private:
   kj::ForkedPromise<void> revokeProm;
 };
 
+// A WorkerInterface that cancels WebSockets when revokeProm is rejected.
+// Currently only supports cancelling for upgrades.
+class RevocableWebSocketWorkerInterface final: public WorkerInterface {
```

Comment:
Thank you for doing some more cleanup!
Since we're add it, with this we can wrap RevocableWebSocketWorkerInterface in an anonymous namespace right? Should slightly improve code size and more importantly warnings.